projects/ngneat/reactive-forms/src/lib/formGroup.spec.ts

```diff
@@ -33,6 +35,68 @@ const createGroup = (withError = false) => {
   );
 };
 
+const createArray = <T>(elements: T[], withError = false): FormArray<T> => {
+  const controlList = elements.map(element => new FormControl<T>(element)) as Array<AbstractControlOf<T>>;
+  return new FormArray<T>(controlList, withError ? errorFn : []);
+};
+
+describe('FormGroup valueChanges$ diff() operator', () => {
+  const control = createGroup();
+  const spy = jest.fn();
+  control.value$.pipe(diff()).subscribe(spy);
+
+  it('should be initialized', () => {
+    expect(spy).toHaveBeenCalledWith({ name: null, phone: { num: null, prefix: null }, skills: [] });
+    expect(spy).toHaveBeenCalledTimes(1);
+  });
+
+  it('should filter duplicated calls', () => {
+    control.patchValue({ name: 'changed' });
+    expect(spy).toHaveBeenCalledWith({ name: 'changed' });
+    control.patchValue({ name: 'changed' });
+    expect(spy).toHaveBeenCalledTimes(2);
+  });
+
+  it('should allow deep FormGroup duplicated calls filtering', () => {
+    control.patchValue({ phone: { num: 1, prefix: 1 } });
+    expect(spy).toHaveBeenCalledWith({ phone: { num: 1, prefix: 1 } });
+    control.patchValue({ phone: { num: 1, prefix: 1 } });
+    expect(spy).toHaveBeenCalledTimes(3);
+  });
+
+  it('should allow deep FormArray duplicated calls filtering', () => {
+    control.setControl('skills', createArray(['driving']));
+    expect(spy).toHaveBeenCalledWith({ skills: ['driving'] });
+    control.setControl('skills', createArray(['driving']));
+    expect(spy).toHaveBeenCalledTimes(4);
+  });
+
```

Comment:
There is an example of pushing into the array?

Reply:
I missed that one, will be added.